### PR TITLE
fix test issues on windows

### PR DESCRIPTION
**Pull request by ph**

### CODE REVIEW

File: \ [hunk 1]
```diff
@@ -0,0 +1,245 @@
+# encoding: UTF-8
```
Comment (jsvd) [inline \]: remove this file :)

File: \ [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+    }
+    CONFIG
+
+    agent do
+      line_num = 0
+      
+      # Now check all events for order and correctness.
+      tmp_file.each_line do |line|
+        event = LogStash::Event.new(LogStash::Json.load(line))
+        insist {event["message"]} == "hello world"
```
Comment (purbon) [inline \]: no need to use insist here ;-), you can use the Rspec expect command ;-) https://relishapp.com/rspec/rspec-expectations/v/2-14/docs

File: \ [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+      end # agent
+    end
+  end
+
+  describe "#register" do
+
+
+    it 'doesnt allow the path to start with a dynamic string' do
+      path = '/%{name}'
+      output = LogStash::Outputs::File.new({ "path" => path })
```
Comment (purbon) [inline \]: This can be generalized in a let variable.

File: \ [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+
+      path = '/a %{name}/'
+      output = LogStash::Outputs::File.new({ "path" => path })
+      expect { output.register }.to raise_error(LogStash::ConfigurationError)
+
+      path = '/a- %{name}/'
+      output = LogStash::Outputs::File.new({ "path" => path })
+      expect { output.register }.to raise_error(LogStash::ConfigurationError)
+
+      path = '/a- %{name}'
```
Comment (purbon) [inline \]: Is this test in pourpouse to repeat the same as in line 97?

File: \ [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+
+
+    it 'doesnt allow the path to start with a dynamic string' do
+      path = '/%{name}'
+      output = LogStash::Outputs::File.new({ "path" => path })
+      expect { output.register }.to raise_error(LogStash::ConfigurationError)
+    end
+
+    it 'doesnt allow the root directory to have some dynamic part' do
+      path = '/a%{name}/'
```
Comment (purbon) [inline \]: This are really three different test cases of the same kind of test, we should DRY it a bit..

File: \ [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+      output = LogStash::Outputs::File.new({ "path" => path })
+      expect { output.register }.to raise_error(LogStash::ConfigurationError)
+
+      path = '/a- %{name}'
+      expect { output.register }.to raise_error(LogStash::ConfigurationError)
+    end
+
+    it 'allow to have dynamic part after the file root' do
+      path = '/tmp/%{name}'
+      output = LogStash::Outputs::File.new({ "path" => path })
```
Comment (purbon) [inline \]: Same things are before, should be DRYed out.

File: \ (hunk 1 above)
Comment (ph) [inline \]: No i wont remove this! I should add this one too to my global gitignore.

File: \ (hunk 2 above)
Comment (ph) [inline \]: This file should not exist, my mistake.. The comment should  be made on the actual diff.

Well, I agree we need to use the `expect` syntax, but inside an agent`s block the expect method isn`t available.
Case like this should be rewritten once we have a new testing strategy.

```
     Failure/Error: block.call
     NoMethodError:
       undefined method `expect' for RSpec::Core::ExampleGroup::Nested_1::Nested_1:Class
```

File: \ (hunk 4 above)
Comment (ph) [inline \]: the path is different, one you are explicitly declaring the `/` and not the other.

File: \ (hunk 3 above)
Comment (ph) [inline \]: Didnt know that, you still have to declare path as a let variable to be able to override it.

**Comment by ph** [comment]
@purbon Followed a few of you tips, Thanks! :)
Updated @jsvd

**Comment by jsvd** [comment]
Depends on https://github.com/elasticsearch/logstash/issues/2597 to work properly, but in itself LGTM

**Comment by jsvd** [comment]
applying the monkeypatch and current master, this plugin's specs pass on windows.

**Pull request closed without merging**